### PR TITLE
add post about docker credentials store

### PR DESCRIPTION
**Pull request by runcom**
@rhatdan @simo5 @nalind

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
@rhatdan updated, PTAL

**Comment by rhatdan** [comment]
@runcom I usually open these as issues, but not sure how @jberkus would prefer.

**Comment by runcom** [comment]
@rhatdan ack, I usually write them directly in the admin interface so it generates the markdown there, @jberkus any preference?

**Comment by rhatdan** [comment]
LGTM.

**Comment by simo5** [comment]
LGTM as well

File: source/blog/2016-03-11-docker-credentials-store.html.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+Docker is also providing a library to help building those credentials helpers at [docker/docker-credential-helpers](https://github.com/docker/docker-credential-helpers).
+
+To configure an *external* credentials store a user need an external helper program to interact with a specific keychain or external store. Docker requires the helper  program to be in the client's host `$PATH`. Docker itself is providing sane defaults for osx and windows and I've recently added support for a credentials store for linux ([docker/docker-credential-helpers#7](https://github.com/docker/docker-credential-helpers/pull/7)) which is using `libsecret` to work with the gnome keyring (and the KDE one). 
+
+Once the helper program is installed on the client's host the user just needs to tell the Docker client to actually use it. Beaware the user needs to perform a `docker logout` to remove the credentials already stored in the json file.
```
Comment (nalind) [inline source/blog/2016-03-11-docker-credentials-store.html.md]: Typo: Beaware

File: source/blog/2016-03-11-docker-credentials-store.html.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+modified = 2016-03-12 12:29:31
+schema = io.docker.Credentials
+attribute.username = testuser
+attribute.server = localhost:5001
+attribute.docker_cli = 1
+```
+
+When interacting with registries Docker queries the credentials store configured and uses the credentials returned (if any).
+
+Generally, it's desiderable to have the system credentials store managing secrets (specifically passwords) in order to apply any additional security measures (apart from file permissions).
```
Comment (nalind) [inline source/blog/2016-03-11-docker-credentials-store.html.md]: Typo: desiderable

**Comment by jberkus** [comment]
Questions:

"Docker 1.11 will be able to automatically detect and use a default credential
store for the platform it will run on"

I'm trying to make sense of this sentence so I can correct it.  If it's a default credential store, why does Docker need to detect it?


"json" is an acronym: JSON



"The implementation shells out to a helper program when a credentials store is configured"

Shells out?  Really?  It doesn't call a subprocess or similar?  Just making sure.


"I've recently added support for a credentials store for linux ([docker/docker-credential-helpers#7](https://github.com/docker/docker-credential-helpers/pull/7)) which is using `libsecret` to work with the gnome keyring (and the KDE one)."

Presumably this is for Desktop Linux only.  What should users do for server Linux?  Obviously it's not implemented, but do we have ideas?

**Comment by runcom** [comment]
> Presumably this is for Desktop Linux only. What should users do for server Linux? Obviously it's not implemented, but do we have ideas?

idea here is `custodia` (I've talked about this at the end of the post)

**Comment by runcom** [comment]
> "json" is an acronym: JSON

ack



> "The implementation shells out to a helper program when a credentials store is configured"
> 
> Shells out? Really? It doesn't call a subprocess or similar? Just making sure.

kind of, I think subprocess is better, thx

**Comment by runcom** [comment]
> I'm trying to make sense of this sentence so I can correct it. If it's a default credential store, why does Docker need to detect it?

because it's an external binary so docker first try to lookup the binary in the host, if it exists it will use it, otherwise it keeps using the json file

**Comment by jberkus** [comment]
@rhatdan I prefer PRs, actually, I just thought you preferred to use issues.

**Comment by jberkus** [comment]
Replaced by PR #270 after edits.  Please continue any discussion on that PR.

**Comment by rhatdan** [comment]
@jberkus Your the boss.  I don't care.  For now on they will be PRs.

**Comment by jberkus** [comment]
@rhatdan if issues are a lot easier for you, then do them.  It's like 3 minutes extra effort for me

**Comment by rhatdan** [comment]
No difference to me. Just pasting into one window or another.  Will do pull requests for now on.

**Pull request closed without merging**